### PR TITLE
(GH-701) Stop using scope.lookupvar() in templates

### PR DESCRIPTION
**Pull request by Phil-Friderici**
Add tests for existing functionality as preparation and to proof backward compatibility.

### CODE REVIEW

**Comment by Phil-Friderici** [comment]
I have separated the commits to proof that the existing functionality is not changed. If prefered I can squash them into one commit.

**Comment by Phil-Friderici** [comment]
ready for review

**Comment by ghoneycutt** [comment]
Please squash into one commit. Thanks!

**Comment by jeffmccune** [comment]
This looks good to me.  The tests are great!  Only question is why scope.lookup var is being left in `sensu-client.erb` and `sensu-windows-package-version.erb`?

**Comment by Phil-Friderici** [comment]
Thanks for the positive feedback, will do the same for the other templates as well :)

**Comment by Phil-Friderici** [comment]
Here we go:
- all appearances of scope.lookupcvar() have been removed
- all commits squashed into one

File: manifests/package.pp [hunk 1]
```diff
@@ -49,7 +108,7 @@
       # $pkg_version is passed to Package[sensu] { ensure }.  The Windows MSI
       # provider translates hyphens to dots, e.g. '0.29.0-11' maps to
       # '0.29.0.11' on the system.  This mapping is necessary to converge.
-      $pkg_version = template('sensu/sensu-windows-package-version.erb')
+      $pkg_version = regsubst($::sensu::version, '-', '.')
```
Comment (ghoneycutt) [inline manifests/package.pp]: good catch!

**Comment by ghoneycutt** [comment]
@Phil-Friderici looks like we still have a few

```
sensu-puppet$ grep -ri scope.lookupvar .
./manifests/extension.pp:  $filename = inline_template('<%= scope.lookupvar(\'source\').split(\'/\').last %>')
./manifests/handler.pp:    $filename = inline_template('<%= scope.lookupvar(\'source\').split(\'/\').last %>')
./manifests/mutator.pp:    $filename = inline_template('<%= scope.lookupvar(\'source\').split(\'/\').last %>')
./manifests/plugin.pp:      $filename = inline_template('<%= scope.lookupvar(\'name\').split(\'/\').last %>')
./manifests/plugin.pp:      $filename = inline_template('<%= scope.lookupvar(\'name\').split(\'/\').last %>')
```

While you are changing these, could you have them use .erb files. I prefer to use .erb files, even for one liners, so they can be validated.

File: manifests/package.pp (hunk 1 above)
Comment (Phil-Friderici) [inline manifests/package.pp]: same is true for sensu-version.erb. I have created the issue https://github.com/sensu/sensu-puppet/issues/731 to get it changed later on too.

**Comment by Phil-Friderici** [comment]
have added the grep results to https://github.com/sensu/sensu-puppet/issues/731 as I think they should be refactored to Puppet code instead.

**Comment by ghoneycutt** [comment]
Yes, please address that in this PR.

Solves #731

**Comment by ghoneycutt** [comment]
@Phil-Friderici please use vagrant to confirm that all platforms still work. Here's a bash function I use to see what clients are connected, instead of poking around through the Dashboard.

```bash
function sensuclients {
  curl -s http://admin:secret@127.0.0.1:4567/clients | jq .[].name | awk -F \" '{print $2}' | awk -F \. '{print $1}' | sort
}
```

**Comment by Phil-Friderici** [comment]
Think we should replace Ruby with Puppet code where possible. So I have replaced all occurrences of
```inline_template('<%= scope.lookupvar(\'source\').split(\'/\').last %>')``` with the much simplier to read ```basename()``` function.

Sorry, haven't used vagrant yet. Will need some time to get into the details and understand it.

File: manifests/package.pp [hunk 2]
```diff
@@ -69,7 +128,7 @@
         # The OS Release specific sub-folder
         $os_release = $facts['os']['release']['major']
         # e.g. '2012 R2' => '2012r2'
-        $pkg_url_dir = template('sensu/sensu-version.erb')
+        $pkg_url_dir = regsubst($os_release, '^(\d+)\s*[rR](\d+)', '\\1r\\2')
```
Comment (Phil-Friderici) [inline manifests/package.pp]: same as for line 111 above

**Comment by Phil-Friderici** [comment]
sweet :)

**Comment by ghoneycutt** [comment]
Released in v2.22.0

**Pull request merged**